postcard_creator/postcard_creator_swissid.py

```diff
@@ -35,3 +64,62 @@ def _do_op(self, method, endpoint, **kwargs):
             e.server_response = response.text
             raise e
         return response
+
+    def _validate_model_response(self, endpoint, payload):
+        if payload.get('errors'):
+            raise PostcardCreatorException(f'cannot fetch {endpoint}: {payload["errors"]}')
+
+    def get_quota(self):
+        logger.debug('fetching quota')
+        endpoint = '/user/quota'
+
+        payload = self._do_op('get', endpoint).json()
+        self._validate_model_response(endpoint, payload)
+        return payload['model']
+
+    def has_free_postcard(self):
+        return self.get_quota()['available']
+
+    def get_user_info(self):
+        logger.debug('fetching user information')
+        endpoint = '/user/current'
+
+        payload = self._do_op('get', endpoint).json()
+        self._validate_model_response(endpoint, payload)
+        return payload['model']
+
+    def get_billing_saldo(self):
+        logger.debug('fetching billing saldo')
+        endpoint = '/billingOnline/accountSaldo'
+
+        payload = self._do_op('get', endpoint).json()
+        self._validate_model_response(endpoint, payload)
+        return payload['model']
+
+    @_send_free_card_defaults
+    def send_free_card(self, postcard, mock_send=True, **kwargs):
+        if not self.has_free_postcard():
+            raise PostcardCreatorException('Limit of free postcards exceeded. Try again tomorrow at '
+                                           + self.get_quota()['next'])
+        if not postcard:
+            raise PostcardCreatorException('Postcard must be set')
+        postcard.validate()
+
+        endpoint = '/card/upload'
+        payload = {
```

Comment:
I also tried something yesterday, but without success so far. It seems that there are three fields. `textImage`, `text` and `image`. The `image`and `textImage` fields are coded images. The `text` field is empty when sending from the App; I think either the text is sent as an image or as plain text.

Reply:
text is empty. it's possible that it is used as a legacy way to submit text if the client does not support drawing text on a canvas. will see if it's supported once the printed out card arrives.